File: boj/2920.음계.py
data = list(map(int, input().split()))

output = "mixed"
for x in range(len(data)-1):
    if data[x] == data[x+1] - 1:
        output = "ascending"
    elif data[x] == data[x+1] + 1:
        output = "descending"
    else:
        output = "mixed"
        break
print(output)

File: by-kotlin/kt-boj/solutions/2920-음계.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    var ascending = true
    var descending = true

    val nums = readLine().split(" ").map { it.toInt() }

    for (i in 0 until 7) {
        if (nums[i] < nums[i + 1]) descending = false
        else if (nums[i] > nums[i + 1]) ascending = false
    }

    if (ascending) println("ascending")
    else if (descending) println("descending")
    else println("mixed")
}
